### DISCUSSION TITLE
Location estimates - What is the priority order of possible sources?

### DISCUSSION THREAD

**Discussion by dvpfig** [photoprism/photoprism#3142] [Q&A] [answered]
Hi all, 

The location estimates are usually pretty conservative. Typically, I've seen it working without GPS data when there is the name of the country in the folder title but not even always it picks it up from there. 

However, I got once a pretty wrong estimate of the location and I cannot figure out where the guess came from.

All the pictures in this folder are from the same source (same phone) and they are all without timezone or GPS data, as far as I can see from the EXIFTOOL (exiftool -j). The title of the folder contains the capital city name (not the country). 
But I guess there was something weird with the name of the folder that I gave. 

Is there documented somewhere all the sources and weights that photoprism uses for Location Estimates when there is no GPS data? 

I've been trying to add information in EXIF for photoprism to take the cue but it doesn't usually take it. 

Something like these sources, I'm expecting: 

1. Exif meta data - if the fields City/Country are present, then take it as Location. (this does not seem to be happening when I made a few tests with IPTC origin city/country). 

2. Folder name - Check if name of country exists in folder title. or name of known city. I guess this is what this file is for: https://github.com/photoprism/photoprism/blob/develop/pkg/txt/resources/countries.txt
I've seen this happening with country names though not always.

**Comment by graciousgrey** [comment, marked as answer]
As far as I remember, location estimates are derived only from GPS data. If you have an image without GPS data and on the same day (24h) there are other images with GPS data, the GPS information is extrapolated for the image that had none before.

**Comment by dvpfig** [comment]
**First point:**  @graciousgrey, your comment directed me to the cause of the problem. Big thanks!

I received a video from a friend wishing Merry Xmas on 24 December. He was with his family in another country. This video had GPS coordinates. 
What the "location estimates" did was to assign GPS coordinates to my own photos (that did not have any location) but only the photos between days 23 and 25 December i.e. all photos that were within +/- 12 hours of that video from my friend's video.

Maybe a way to avoid this would be to assign GPS coordinates only if the same persons were detected in the other media files. It's probably a common situation that we have pictures from different places taken by different persons in the same date & time. Maybe it's more rare that the same face is in two different locations at the same date & time.  
I understand the difficulty to solve, so I don't expect a solution. 

But the clear information that @graciousgrey wrote before about "location estimates" should be included here in the docs maybe under a sub-item "location estimates": https://docs.photoprism.app/user-guide/library/metadata/#enrichment

Until now, I thought "location estimates" would only do this if the photos (with and without gps data) were placed in the same folder. I never thought it would take locations from photos of the same day placed under another separate folder (in this case, the Whatsapp folder).

**Reply by graciousgrey** [reply]
> Maybe a way to avoid this would be to assign GPS coordinates only if the same persons were detected in the other media files. It's probably a common situation that we have pictures from different places taken by different persons in the same date & time.

For now it is only possible to turn estimates off. We know that estimates can be further improved and we do already have an [open issue](https://github.com/photoprism/photoprism/issues/622) for it.

As for the documentation - would it be more clear if we add a link from https://docs.photoprism.app/user-guide/library/metadata/#enrichment to https://docs.photoprism.app/user-guide/settings/library/#estimates ?

**Reply by dvpfig** [reply]
> > Maybe a way to avoid this would be to assign GPS coordinates only if the same persons were detected in the other media files. It's probably a common situation that we have pictures from different places taken by different persons in the same date & time.
> 
> For now it is only possible to turn estimates off. We know that estimates can be further improved and we do already have an [open issue](https://github.com/photoprism/photoprism/issues/622) for it.
> 
> As for the documentation - would it be more clear if we add a link from https://docs.photoprism.app/user-guide/library/metadata/#enrichment to https://docs.photoprism.app/user-guide/settings/library/#estimates ?

Yes, an hyperlink to the description of "estimates" would help in the Enrichment section. 

But I would also add a note in the description of "estimates", e.g. 
"Note: be aware that, if you have pictures from events happening in the same day but different locations, the GPS coordinates of pictures from one event will be applied to pictures of the other event that lack coordinates."

**Reply by lastzero** [reply]
It's not just about the same day, but the nearest picture in time and your travel velocity.

**Reply by lastzero** [reply]
Note you can easily send documentation improvements by clicking the edit link in the upper right corner.

**Reply by dvpfig** [reply]
Yes @lastzero, will do the proposal to edit the documentation there instead. 
I understand how the GPS estimate tool works much better now. Thanks to both.

**Comment by dvpfig** [comment]
**Second point:**
I'm pretty sure there is another way that "Location Estimates" adds Place information, though not GPS data points. Example below:

I have a folder that contains only photos from a camera without GPS.
This folder has a country name in its title ("Finland"). 
What the "location estimates" did (or another enrichment tool did) was to assign a place name using the country name. 

The photo doesn't show up in the map of Places (neither in list of States), what it shows is a country name when seeing the picture: 
![bild](https://user-images.githubusercontent.com/39123300/214579054-5a37fc43-4187-4151-a29e-0d7f0a6434e7.png)



Below is the sidecar of one photo 

```
TakenAt: 2018-08-11T09:33:22Z
TakenSrc: meta
UID: prot3hs1fbi9z1sp
Type: image
Title: Finland Album / 2018
PlaceSrc: estimate
Year: 2018
Month: 8
Day: 11
ISO: 100
Exposure: 1/125
FNumber: 8
FocalLength: 24
Quality: 3
Details:
  Keywords: album, albums, finland, green
  Software: Adobe Photoshop Lightroom Classic 7.4 (Windows)
  SoftwareSrc: meta
CreatedAt: 2023-01-20T22:51:28Z
UpdatedAt: 2023-01-23T20:59:45Z
EditedAt: 2023-01-23T20:59:45Z
```

**Reply by graciousgrey** [reply]
This is a different mechanism. If a photo without GPS information has a keyword matching a country name it gets this country set as country, it does not get GPS coordinates and therefore cannot be displayed on the map.

**Reply by dvpfig** [reply]
Ok, understood. 

May I attempt to complete your description: 
If a photo without GPS information has a keyword matching a country name OR a city name listed in the file below, it gets this country set as country, it does not get GPS coordinates and therefore cannot be displayed on the map.

The file used to attempt to match country names or city names seems to be this one: 
https://github.com/photoprism/photoprism/blob/develop/pkg/txt/resources/countries.txt

I'm saying this, because I have another situation where an album name "Chicago" was marked for Place as "USA", so the mechanism detects other words than country names. 
What puzzles me with this txt file above, is why there was no attempt to normalize at least a bit across the countries listed. 
We have large countries such as France and Brazil with only the name of country (no list of capital city or any other city), and a country like Germany that lists to the level of districts in Berlin and buildings (e.g. Bundestag). 

Is the intention that the community fill in the gaps by proposing a long list of easy additions? I'd be happy to contribute somehow.

**Reply by lastzero** [reply]
The reason for this is that there is only one full time developer, namely me. Since we still don't have enough funds to grow our team, there is only a very limited time available to work on each feature.

**Reply by dvpfig** [reply]
I didn't intend to sound as complaining criticism, apologies. The work done in PhotoPrism is unique and amazingly good tool. 

The interest shown in better understanding the tools and code is to find ways for the community to help/contribute to improve the existing features.
